src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -130,14 +134,85 @@ public static class TagHelperDescriptorFactory
                         assemblyName,
                         attributeDescriptors,
                         attribute,
+                        allowedChildren,
                         typeDesignTimeDescriptor));
         }
 
+        private static IEnumerable<string> GetAllowedChildren(TypeInfo typeInfo, ErrorSink errorSink)
+        {
+            var restrictChildrenAttribute = typeInfo.GetCustomAttribute<RestrictChildrenAttribute>(inherit: true);
```

Comment:
Method should always return `null` if `TagStructure` is `WithoutEndTag` in this descriptor. Also need an error if that tag structure is mixed with the `RestrictChildrenAttribute`. Main question is whether the error should be displayed once per tag helper or once per conflicting `TargetElementAttribute`.